hawkmoth/__init__.py

```diff
@@ -239,6 +239,37 @@ class CAutoEnumDirective(_AutoCompoundDirective):
     _domain = 'c'
     _docstring_types = [docstring.EnumDocstring]
 
+class CppAutoDocDirective(_AutoDocDirective):
+    _domain = 'cpp'
+
+class CppAutoVarDirective(_AutoSymbolDirective):
+    _domain = 'cpp'
+    _docstring_types = [docstring.VarDocstring]
+
+class CppAutoTypeDirective(_AutoSymbolDirective):
+    _domain = 'cpp'
+    _docstring_types = [docstring.TypeDocstring]
+
+class CppAutoMacroDirective(_AutoSymbolDirective):
+    _domain = 'cpp'
+    _docstring_types = [docstring.MacroDocstring, docstring.MacroFunctionDocstring]
```

Comment:
So the idea is we have `.. cpp:automacro:` but it ends up as `.. c:macro:` in reStructuredText?

Reply:
That was my idea, yes. Of course I would hope Sphinx eventually adds `cpp:macro`.

Even if that never happens, my arguments:
1. `cpp:autodoc` will produce `c:macro` anyway.
2. The user is going to bump into the inconsistency when referencing macros either way.
3. This avoids a special case for the domain detection, which is user facing API. (It also simplifies implementation.)